rpcs3/Emu/RSX/VK/vkutils/garbage_collector.h

```diff
@@ -0,0 +1,56 @@
+#include <util/types.hpp>
+
+#include <functional>
+
+namespace vk
+{
+	class disposable_t
+	{
+		void* ptr;
+		std::function<void(void*)> deleter;
+
+		disposable_t(void* ptr_, std::function<void(void*)> deleter_) :
+			ptr(ptr_), deleter(deleter_) {}
+	public:
+
+		disposable_t() = delete;
+		disposable_t(const disposable_t&) = delete;
+
+		disposable_t(disposable_t&& other) :
+			ptr(std::exchange(other.ptr, nullptr)),
+			deleter(other.deleter)
+		{}
+
+		~disposable_t()
+		{
+			if (ptr)
+			{
+				deleter(ptr);

```

Comment:
Deleter may be null

Reply:
This class has no public constructor. That scenario is not possible.